theano/scalar/basic.py

```diff
@@ -3063,14 +3077,20 @@ def init_c_code(self):
                 + zip(self.fgraph.outputs,
                     ["%%(o%i)s" % i for i in xrange(len(self.fgraph.outputs))]))
 
-        for orphan in self.fgraph.variables:  # fgraph.orphans:
-            if orphan.owner is None and orphan not in self.fgraph.inputs:
-                if isinstance(orphan, Constant):
-                    subd[orphan] = orphan.type.c_literal(orphan.data)
-                else:
-                    raise ValueError(
-                        "All orphans in the fgraph to Composite must"
-                        " be Constant instances.")
+        for var in self.fgraph.variables:
+            if var.owner is None:
+                if var not in self.fgraph.inputs:
+                    # This is an orphan
+                    if isinstance(var, Constant):
+                        subd[var] = var.type.c_literal(var.data)
+                    else:
+                        raise ValueError(
+                            "All orphans in the fgraph to Composite must"
+                            " be Constant instances.")
+            elif any(i.dtype == 'float16' for i in var.owner.inputs or
+                     o.dtype == 'float16' for o in var.owner.outputs):
+                # flag for elemwise ops to check.
+                self.inner_float16 = True
```

Comment:
I won't botter with this, but it being not set could mean we didn't check it. So setting false would tell we did the check.